client/signup/config/flows.js

```diff
@@ -226,6 +226,14 @@ const Flows = {
 		// if ( Flow.defaultFlowName === flowName ) {
 		// }
 
+		// Remove About step in the ecommerce flow if we're in the onboarding AB test
+		if ( 'ecommerce' === flowName && 'onboarding' === abtest( 'improvedOnboarding' ) ) {
+			const afterStep = user && user.get() ? '' : 'user';
+
+			flow = Flows.removeStepFromFlow( 'about', flow );
+			return Flows.insertStepIntoFlow( 'site-type', flow, afterStep );
+		}
+
```

Comment:
Do you think it would make sense to redirect users to the `ecommerce-onboarding` flow instead of dynamically updating the `ecommerce` flow? One downside is that it would change the URL to `/start/ecommerce-onboarding/domains`

Reply:
A few reasons why I chose to not redirect the user to a new ecommerce flow:
1) As you rightly said,  the URL will change to `ecommerce-onboarding` flow. 
2) The flow name `ecommerce` is hardcoded in a few places ([#](https://github.com/Automattic/wp-calypso/blob/affe06d124b57f4aac9e2d4e935f200692cd4c78/client/signup/steps/plans/index.jsx#L62), [#](https://github.com/Automattic/wp-calypso/blob/9ede582d81a4641524328ebe4bbd27be37c020ff/client/signup/utils.js#L210), [#](https://github.com/Automattic/wp-calypso/blob/2bc79d736f3bdd6a07374b8ed66c33af72002335/client/signup/steps/domains/index.jsx#L358), [#](https://github.com/Automattic/wp-calypso/blob/c46187dfab764fbe1cc122d8a30dc9faded5b162/client/signup/steps/about/index.jsx#L243)). So, adding a new flow will require changes across more files. The present fix touches the least number of places.
3) Future proofing: if we redirect users to the `ecommerce-onboarding` flow, then we'll need to maintain two flows - `ecommerce` and `ecommerce-onboarding` - in all future additions to the codebase that refer to flowName. I think this makes it more prone to errors. In the present scenario, it's future proofed - all code references to flowName can remain only to `ecommerce`.

Are there any downsides to the approach of dynamically updating the `ecommerce` flow that I haven't considered? Or reasons why it's less preferred?